test/integration/controller_instance_test.go

```diff
@@ -1112,71 +1116,115 @@ func FlakeTestCreateServiceInstanceWithProvisionFailure(t *testing.T) {
 						}
 					}
 					ct.osbClient.ProvisionReaction = fakeosb.DynamicProvisionReaction(
-						getProvisionResponseByPollCountReactions(2, []fakeosb.ProvisionReaction{
-							fakeosb.ProvisionReaction{
-								Error: reactionError,
-							},
-							fakeosb.ProvisionReaction{
-								Response: &osb.ProvisionResponse{},
-							},
-						}))
+						func(r *osb.ProvisionRequest) (*osb.ProvisionResponse, error) {
+							if atomic.LoadInt32(&respondWithProvisionSuccess) != 0 {
```

Comment:
would prefer if this was `== 1`. we set it explicitly up above and I find equality easier to read than inequality.